sdk/core/azure-core/src/main/java/com/azure/core/models/spatial/Geometry.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models.spatial;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * An abstract representation of a geometry.
+ */
+public abstract class Geometry {
+    private final GeometryBoundingBox boundingBox;
+    private final Map<String, Object> properties;
+
+    Geometry(GeometryBoundingBox boundingBox, Map<String, Object> properties) {
+        this.boundingBox = boundingBox;
+
+        if (properties == null) {
+            this.properties = Collections.emptyMap();
+        } else {
+            this.properties = properties.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
Collections.unmodifiableMap?

Reply:
Good catch, I'll fix this in the next update when I add unit tests.